src/ProxyClient/HttpDispatcher.php

```diff
@@ -85,14 +87,15 @@ class HttpDispatcher implements Dispatcher
      *                                         absolute URLs
      * @param HttpAsyncClient|null $httpClient Client capable of sending HTTP requests. If no
      *                                         client is supplied, a default one is created
-     * @param UriFactory|null      $uriFactory Factory for PSR-7 URIs. If not specified, a
+     * @param UriFactoryInterface|null      $uriFactory Factory for PSR-7 URIs. If not specified, a
      *                                         default one is created
      */
     public function __construct(
         array $servers,
         $baseUri = '',
         HttpAsyncClient $httpClient = null,
-        UriFactory $uriFactory = null
+        Psr17FactoryDiscovery $uriFactory = null,
```

Comment:
and this needs a phpdoc `UriFactory|Psr17FactoryDiscovery` and our code needs to handle both cases